tests/test_mc_simulator.py

```diff
@@ -0,0 +1,303 @@
+
+    expected_command = "bad_command"
+    test_item = {"command": expected_command}
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_item, testing_queue
+    )
+    with pytest.raises(UnrecognizedSimulatorTestCommandError, match=expected_command):
+        invoke_process_run_and_check_errors(simulator)
+
+
+def test_MantarrayMCSimulator__handles_reads_of_size_less_than_next_packet_in_queue__when_queue_is_not_empty(
+    mantarray_mc_simulator,
+):
+    _, output_queue, _, _, simulator = mantarray_mc_simulator
+
+    test_size_diff = 2
+    item_1 = b"item_one"
+    item_2 = b"second_item"
+    output_queue.put(item_1)
+    output_queue.put(item_2)
+    confirm_queue_is_eventually_of_size(output_queue, 2)
+
+    expected_1 = item_1[:-test_size_diff]
+    actual_1 = simulator.read(len(item_1) - test_size_diff)
+    assert actual_1 == expected_1
+
+    expected_2 = item_1[-test_size_diff:] + item_2[:-test_size_diff]
+    actual_2 = simulator.read(len(item_2))
+    assert actual_2 == expected_2
+
+    expected_3 = item_2[-test_size_diff:]
+    actual_3 = simulator.read(
+        len(expected_3)
+    )  # specifically want to read exactly the remaining number of bytes
+    assert actual_3 == expected_3
+
+
+def test_MantarrayMCSimulator__handles_reads_of_size_less_than_next_packet_in_queue__when_queue_is_empty(
+    mantarray_mc_simulator,
+):
+    _, output_queue, _, _, simulator = mantarray_mc_simulator
+
+    test_size_diff_1 = 3
+    test_size_diff_2 = 1
+    test_item = b"first_item"
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        test_item, output_queue
+    )
+
+    expected_1 = test_item[:-test_size_diff_1]
+    actual_1 = simulator.read(len(test_item) - test_size_diff_1)
+    assert actual_1 == expected_1
+
+    expected_2 = test_item[-test_size_diff_1:-test_size_diff_2]
+    actual_2 = simulator.read(test_size_diff_1 - test_size_diff_2)
+    assert actual_2 == expected_2
+
+    expected_3 = test_item[-1:]
+    actual_3 = simulator.read(1)
+    assert actual_3 == expected_3
+
+
+@pytest.mark.slow
+def test_MantarrayMCSimulator__handles_reads_of_size_less_than_next_packet_in_queue__when_simulator_is_running(
+    mantarray_mc_simulator,
+):
+    _, output_queue, _, testing_queue, simulator = mantarray_mc_simulator
+
+    test_item_1 = b"12345"
+    test_item_2 = b"67890"
+    testing_queue.put({"command": "add_read_bytes", "read_bytes": test_item_1})
+    testing_queue.put({"command": "add_read_bytes", "read_bytes": test_item_2})
+    confirm_queue_is_eventually_of_size(testing_queue, 2)
+
+    # Tanner (2/2/20): this try/finally block ensures that the simulator is stopped even if the test fails. Problems can arise from processes not being stopped after tests complete
+    try:
+        simulator.start()
+        confirm_queue_is_eventually_empty(  # Tanner (2/2/21): Adding a large timeout here to avoid sporadic failures in CI (windows did not like a 5 second timeout)
```

Comment:
if this is causing problems and it's not critical to this test, you could `invoke_process_run` twice prior to starting the simulator (assuming the goal of this test is just focused on the `.read` function and not the ability of the simulator to be populated with new data while runnning)